src/TranslationFormTrait.php

```diff
@@ -41,6 +50,18 @@ protected function translationFormElement(array $data, array $existing_translati
       // The char sequence '][' confuses the form API so we need to replace
       // it.
       $target_key = str_replace('][', '|', $key);
+
+      if (isset($data[$key]['#remove'])) {
+        $element[$target_key] = [
+          '#theme' => 'status_messages',
+          '#message_list' => [
+            'warning' => [$this->t('Source value for the field <em>@name</em> was removed. Synchronising this translation will remove the value from the translation as well.', ['@name' => $data[$key]['#parent_label'][0]])],
```

Comment:
When removing the value of a paragraph field `$data[$key]['#parent_label'][0]` contains the entity reference revisions field label.

When multiple fields had their values removed, the warning mentions only the last field.